addons/website_mass_mailing/models/mass_mailing_list.py

```diff
@@ -1,23 +1,66 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import fields, models, api
 
 
 class MassMailingList(models.Model):
     _inherit = 'mail.mass_mailing.list'
 
+    def _default_website(self):
+        return self.env['website'].get_current_website()
+
     def _default_popup_content(self):
-        return """<div class="modal-header text-center">
-    <h3 class="modal-title mt8">Odoo Presents</h3>
-</div>
-<div class="o_popup_message">
-    <font>7</font>
-    <strong>Business Hacks</strong>
-    <span> to<br/>boost your marketing</span>
-</div>
-<p class="o_message_paragraph">Join our Marketing newsletter and get <strong>this white paper instantly</strong></p>"""
-
-    popup_content = fields.Html(string="Website Popup Content", translate=True, sanitize_attributes=False,
-                                default=_default_popup_content)
-    popup_redirect_url = fields.Char(string="Website Popup Redirect URL", default='/')
+        return self.env['website'].get_template('website_mass_mailing.s_newsletter_popup_template').render()
+
+    website_popup_ids = fields.One2many('mail.mass_mailing.popup', 'mailing_list_id')
+    website_id = fields.Many2one('website', string="Website", default=_default_website)
+    current_popup_id = fields.Many2one('mail.mass_mailing.popup', compute="_compute_current_popup")
```

Comment:
I am not sure to understand your model here. You have a mailing list which is website specific at all times but which can contains popup from all website ? ... seems strange

A mailing list should not be website dependent, only the popups.